File: heimdall.sample-client/src/test/java/com/miqdigital/cucumber_runner/javasample/KarateTestRunner.java
package com.miqdigital.cucumber_runner.javasample;

import static org.junit.Assert.assertEquals;

import java.io.IOException;

import org.junit.Test;

import com.intuit.karate.KarateOptions;
import com.intuit.karate.Results;
import com.intuit.karate.Runner;
import com.miqdigital.Heimdall;

/**
 * The type Karate test runner.
 */
@KarateOptions(features = {"classpath:features/Karate"},
               tags = {"@regression"})
public class KarateTestRunner {


  private final String pathOfPropertyFile = "";
  private final String karateOutputPath = "";

  /**
   * Heimdall reporting.
   *
   * @throws InterruptedException the interrupted exception
   * @throws NoSuchFieldException the no such field exception
   * @throws IllegalAccessException the illegal access exception
   * @throws IOException the io exception
   */
  @Test
  public void heimdallReporting()
      throws InterruptedException, NoSuchFieldException, IllegalAccessException, IOException {

    Results results = Runner.path(karateOutputPath).tags("@regression").parallel(10);

    final Heimdall heimdall = new Heimdall();
    heimdall.updateStatusInS3AndNotifySlack(pathOfPropertyFile, karateOutputPath);
    assertEquals("There are scenario failures", 0, results.getFailCount());
  }
}

File: heimdall.sample-client/src/test/java/com/miqdigital/cucumber_runner/kotlinesample/KarateTestRunner.kt
package com.miqdigital.cucumber_runner.kotlinesample

import com.intuit.karate.cucumber.CucumberRunner
import com.miqdigital.Heimdall
import cucumber.api.CucumberOptions
import junit.framework.Assert.assertEquals
import org.junit.Test
import java.io.IOException

/**
 * The type Karate test runner.
 */
@CucumberOptions(features = ["classpath:features/Karate"], tags = ["@regression"])
class KarateTestRunner {


    private val pathOfPropertyFile = ""
    private val cucumberOutputPath = ""

    /**
     * Heimdall reporting.
     *
     * @throws InterruptedException   the interrupted exception
     * @throws NoSuchFieldException   the no such field exception
     * @throws IllegalAccessException the illegal access exception
     * @throws IOException            the io exception
     */
    @Test
    @Throws(InterruptedException::class, NoSuchFieldException::class, IllegalAccessException::class, IOException::class)
    fun heimdallReporting() {

        val karateStats = CucumberRunner.parallel(javaClass, 5, cucumberOutputPath)

        val heimdall = Heimdall()
        heimdall.updateStatusInS3AndNotifySlack(pathOfPropertyFile, cucumberOutputPath)
        assertEquals("There are scenario failures", 0, karateStats.failCount.toLong())
    }
}